core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/upgrade/recover/DatalakeRecoverySetupNewInstancesHandler.java

```diff
@@ -81,7 +81,7 @@ private void setupNewInstances(Stack stack) {
         for (InstanceGroup instanceGroup : instanceGroups) {
             List<CloudInstance> newInstance =
                     stackUpscaleService.buildNewInstances(stack, instanceGroup.getGroupName(), instanceGroup.getInitialNodeCount());
-            instanceMetaDataService.saveInstanceAndGetUpdatedStack(stack, newInstance, true, Collections.emptySet());
+            instanceMetaDataService.saveInstanceAndGetUpdatedStack(stack, newInstance, true, Collections.emptySet(), false);
```

Comment:
Just a sanity check: I am not familiar with the DL upgrade recovery process; does it always discard former volumes when creating replacement instances?

Reply:
It is the original functionality. I will check the DL upgrade in the next phase